runtime/src/iree/modules/hal/debugging.c

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/modules/hal/debugging.h"
+
+//===----------------------------------------------------------------------===//
+// Debug Sink
+//===----------------------------------------------------------------------===//
+
+IREE_API_EXPORT iree_hal_module_debug_sink_t
+iree_hal_module_debug_sink_null(void) {
+  iree_hal_module_debug_sink_t sink = {0};
+  return sink;
+}
+
+static iree_status_t iree_hal_module_buffer_view_trace_stdout(
```

Comment:
Shouldn't this be called `iree_hal_module_buffer_view_trace_stderr`? It actually writes to `stderr`.

Reply:
good catch!

Reply:
I renamed it to stdio and it now takes a file handle so the caller gets to decide :)